src/pyk/proof/reachability.py

```diff
@@ -717,18 +735,21 @@ def step_proof(self) -> Iterable[StepResult]:
         if self.proof.bmc_depth is not None and curr_node.id not in self._checked_for_bounded:
             _LOGGER.info(f'Checking bmc depth for node {self.proof.id}: {curr_node.id}')
             self._checked_for_bounded.add(curr_node.id)
-            _prior_loops = [
-                succ.source.id
-                for succ in self.proof.shortest_path_to(curr_node.id)
-                if self.kcfg_explore.kcfg_semantics.same_loop(succ.source.cterm, curr_node.cterm)
-            ]
-            prior_loops: list[NodeIdLike] = []
-            for _pl in _prior_loops:
-                if not (
-                    self.proof.kcfg.zero_depth_between(_pl, curr_node.id)
-                    or any(self.proof.kcfg.zero_depth_between(_pl, pl) for pl in prior_loops)
-                ):
-                    prior_loops.append(_pl)
+
+            prior_loops = []
+            for succ in reversed(self.proof.shortest_path_to(curr_node.id)):
+                if self.kcfg_explore.kcfg_semantics.same_loop(succ.source.cterm, curr_node.cterm):
+                    if succ.source.id in self.proof.prior_loops_cache.keys():
```

Comment:
```suggestion
                    if succ.source.id in self.proof.prior_loops_cache:
```

`_in_` is implemented directl yfor dictionaries.